src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -277,6 +277,99 @@ void GraphOptimizer::FuseConvMatmulFCDeconvAndDQScales(Graph &graph) {
     }
 }
 
+void GraphOptimizer::FuseFCAndWeightsDecompression(Graph &graph) {
+    auto expectedNode = [](NodePtr node, Type expectedType) {
+        return node->getType() == expectedType && node->getChildEdges().size() == 1;
+    };
+
+    auto& graphNodes = graph.GetNodes();
+    for (size_t i = 0; i < graphNodes.size(); i++) {
+        const auto fcNode = dynamic_cast<node::FullyConnected*>(graphNodes[i].get());
+        if (fcNode == nullptr)
+            continue;
+
+        const auto parent = fcNode->getParentEdgesAtPort(1)[0]->getParent();
+        const bool with_transpose = parent->getType() == Type::Transpose;
+        const NodePtr transposeNode = with_transpose ? parent : nullptr;
+
+        const auto multiplyNode = with_transpose ? parent->getParentEdgesAtPort(0)[0]->getParent() : parent;
+        if (!expectedNode(multiplyNode, Type::Eltwise) || multiplyNode->getAlgorithm() != Algorithm::EltwiseMultiply ||
+            !multiplyNode->isConstant())
+            continue;
+
+        const auto multiplyConstNode = multiplyNode->getParentEdgesAtPort(1)[0]->getParent();
+        if (!expectedNode(multiplyConstNode, Type::Input))
+            continue;
+
```

Comment:
Would propose to add `CPU_GRAPH_OPTIMIZER_SCOPE(FuseFCAndWeightsDecompression);` line here to enable conditional compilation optimization for this pass.

Reply:
Added, thanks